File: src/main/java/com/ivanqueiroz/kotlin/Principal.java
package com.ivanqueiroz.kotlin;

import com.ivanqueiroz.kotlin.dto.Usuario;

/**
 *
 * @author Ivan Queiroz <ivanqueiroz@gmail.com>
 */
public class Principal {
    
    public static void main(String[] args) {
        Usuario usuario = new Usuario("JOAO", 35);
        Usuario usuario2 = new Usuario("MARIA", 30);
        
        System.out.println("Nome: "+usuario.getNome());
        System.out.println("Idade: "+usuario.getIdade());
        System.out.println(usuario.toString());
        
        System.out.println("Nome: "+usuario2.getNome());
        System.out.println("Idade: "+usuario2.getIdade());
        System.out.println(usuario2.toString());
        
        System.out.println("Hash usuario: "+usuario.hashCode());
        System.out.println("Hash usuario2: "+usuario.hashCode());
        
        System.out.println("usuario.equals(usuario2) = "+usuario.equals(usuario2));
        
    }
    
}

File: src/main/java/com/ivanqueiroz/kotlin/Principal.kt
package com.ivanqueiroz.kotlin

import com.ivanqueiroz.kotlin.dto.Usuario

/*

  @author Ivan Queiroz <ivanqueiroz@gmail.com>
  Created on 15/01/2017
*/

fun main(args: Array<String>) {
    val usuario = Usuario("João", 35)
    val usuario2 = Usuario("Maria", 35)

    println("Nome: ${usuario.nome}")
    println("Idade: ${usuario.idade}")
    println(usuario.toString())
    println("Hashcode: ${usuario.hashCode()}")

    println("Nome: ${usuario2.nome}")
    println("Idade: ${usuario2.idade}")
    println(usuario2.toString())
    println("Hashcode: ${usuario2.hashCode()}")

    println("usuario.equals(usuario2) = ${usuario.equals(usuario2)}")

    val usuario3: Usuario? = Usuario("Pedro", 30)
    if (usuario3 != null) {
        println(usuario3.nome)
    }
}

